Monorepo: Node versions CI tests fail

Currently our [Node versions CI tests](https://github.com/ethereumjs/ethereumjs-monorepo/blob/master/.github/workflows/node-versions.yml) fail for Node `12` and `14`, see e.g. the following [CI run](https://github.com/ethereumjs/ethereumjs-monorepo/actions/runs/1715643832).

This is the associated error message:

```shell
> root@ preinstall /home/runner/work/ethereumjs-monorepo/ethereumjs-monorepo
> npm run checkNpmVersion


> root@ checkNpmVersion /home/runner/work/ethereumjs-monorepo/ethereumjs-monorepo
> ./scripts/check-npm-version.sh

npm version 7 or greater is required for workspaces support. Please update with "npm i -g npm@7"
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! root@ checkNpmVersion: `./scripts/check-npm-version.sh`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the root@ checkNpmVersion script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm WARN Local package.json exists, but node_modules missing, did you mean to install?

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/runner/.npm/_logs/2022-01-19T00_59_38_182Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! root@ preinstall: `npm run checkNpmVersion`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the root@ preinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm WARN Local package.json exists, but node_modules missing, did you mean to install?

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/runner/.npm/_logs/2022-01-19T00_59_38_235Z-debug.log
Error: Process completed with exit code 1.
```

This should be fixed. 😄

--- a/.github/workflows/node-versions.yml
+++ b/.github/workflows/node-versions.yml
@@ -8,6 +8,8 @@ name: Node versions
 on:
   schedule:
     - cron: 0 0 * * *
+  pull_request:
+    types: [opened, reopened, synchronize]
 
 jobs:
   get-node-versions:

--- a/.github/workflows/node-versions.yml
+++ b/.github/workflows/node-versions.yml
@@ -8,8 +8,6 @@ name: Node versions
 on:
   schedule:
     - cron: 0 0 * * *
-  pull_request:
-    types: [opened, reopened, synchronize]
 
 jobs:
   get-node-versions:

--- a/.github/workflows/node-versions.yml
+++ b/.github/workflows/node-versions.yml
@@ -40,6 +40,9 @@ jobs:
           node-version: ${{ matrix.node }}
           cache: 'npm'
 
+      - name: Use npm v7 for workspaces support
+        run: npm i -g npm@7
+
       - run: npm i
 
       - name: Test Block

--- a/.github/workflows/node-versions.yml
+++ b/.github/workflows/node-versions.yml
@@ -42,6 +42,7 @@ jobs:
 
       - name: Use npm v7 for workspaces support
         run: npm i -g npm@7
+        if: ${{ matrix.node-version < 16 }}
 
       - run: npm i
 

--- a/packages/rlp/bin/rlp
+++ b/packages/rlp/bin/rlp
@@ -24,7 +24,8 @@ if (method === 'encode') {
     const encoded = RLP.encode(json)
     console.log('0x' + bytesToHex(Uint8Array.from(encoded)))
   } catch (error) {
-    console.error(`Error encoding RLP: ${error.message ?? error}`)
+    const errorMsg = error.message ? error.message : error
+    console.error(`Error encoding RLP: ${errorMsg}`)
   }
 } else if (method === 'decode') {
   // Decode
@@ -33,7 +34,8 @@ if (method === 'encode') {
     const json = JSON.stringify(arrToJSON(decoded))
     console.log(json)
   } catch (error) {
-    console.error(`Error decoding RLP: ${error.message ?? error}`)
+    const errorMsg = error.message ? error.message : error
+    console.error(`Error decoding RLP: ${errorMsg}`)
   }
 } else {
   console.error('Unsupported method')

--- a/packages/rlp/bin/rlp
+++ b/packages/rlp/bin/rlp
@@ -16,13 +16,14 @@ if (method === 'encode') {
   try {
     json = JSON.parse(strInput)
   } catch (error) {
-    return console.error(`Error could not parse JSON: ${JSON.stringify(error)}`)
+    const errorMsg = error.message ? error.message : error
+    return console.error(`Error could not parse JSON: ${errorMsg}`)
   }
 
   // Encode RLP
   try {
     const encoded = RLP.encode(json)
-    console.log('0x' + bytesToHex(Uint8Array.from(encoded)))
+    console.log('0x' + bytesToHex(encoded))
   } catch (error) {
     const errorMsg = error.message ? error.message : error
     console.error(`Error encoding RLP: ${errorMsg}`)

